hydrolib/core/io/friction/models.py

```diff
@@ -0,0 +1,145 @@
+    class Comments(INIBasedModel.Comments):
+        frictionid: Optional[str] = Field(
+            "Name of the roughness variable.", alias="frictionId"
+        )
+        frictiontype: Optional[str] = Field(
+            "The global roughness type for this variable, which is used "
+            + "if no branch specific roughness definition is given.",
+            alias="frictionType",
+        )
+        frictionvalue: Optional[str] = Field(
+            "The global default value for this roughness variable.",
+            alias="frictionValue",
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["Global"] = "Global"
+    frictionid: str = Field(alias="frictionId")
+    frictiontype: str = Field(alias="frictionType")
+    frictionvalue: float = Field(alias="frictionValue")
+
+
+class FrictBranch(INIBasedModel):
+    class Comments(INIBasedModel.Comments):
+        branchid: Optional[str] = Field("The name of the branch.", alias="branchId")
+        frictiontype: Optional[str] = Field(
+            "The roughness type to be used on this branch.", alias="frictionType"
+        )
+        functiontype: Optional[str] = Field(
+            "Function type for the calculation of the value. "
+            + "Possible values: constant, timeSeries, absDischarge, waterlevel.",
+            alias="functionType",
+        )
+        timeseriesid: Optional[str] = Field(
+            "Refers to a data block in the <*.bc> frictionValuesFile. "
+            + "Only if functionType = timeSeries.",
+            alias="timeSeriesId",
+        )
+        numlevels: Optional[str] = Field(
+            "Number of levels in table. Only if functionType is not constant.",
+            alias="numLevels",
+        )
+        levels: Optional[str] = Field(
+            "Space separated list of discharge [m3/s] or water level [m AD] values. "
+            + "Only if functionType is absDischarge or waterLevel."
+        )
+        numlocations: Optional[str] = Field(
+            "Number of locations on branch. The default 0 implies branch uniform values.",
+            alias="numLocations",
+        )
+        chainage: Optional[str] = Field(
+            "Space separated list of locations on the branch [m]. Locations sorted by "
+            + "increasing chainage. The keyword must be specified if numLocations>0."
+        )
+        frictionvalues: Optional[str] = Field(
+            "numLevels lines containing space separated lists of roughness values: "
+            + "numLocations values per line. If the functionType is constant, then a "
+            + "single line is required. For a uniform roughness per branch "
+            + "(numLocations = 0) a single entry per line is required. The meaning "
+            + "of the values depends on the roughness type selected (see frictionType).",
+            alias="frictionValues",
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["Branch"] = "Branch"
```

Comment:
Isnt the header name "BranchProperties"?

Reply:
No, it's "Branch", see table C.19 in https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#section.C.15